### DISCUSSION TITLE
FordPass App öffnen bei "Klick" ?

### DISCUSSION THREAD

**Discussion by ChrisWUP** [dschablowsky/FordPassWidget#4] [Ideas]
Hallo Damian,

viele Dank für die coole App :-)
Wirklich schön. ich hätte 2 Ideen / Anmerkungen

a.) Wie kann ich einstellen das sich die ForPass APP öffnen wenn ich das Widget (oder das Ford Logo im Widget) anklicke ?
b.) Lässt sich der Status "Fenster" auch auf alle Türen und Fenster anpassen ?
Du hattest in der Readme gesagt das es möglich wäre einzelne Bereich anzupassen, das habe ich aber nicht geschafft ...

Super APP ! Danke dafür !
Dir einen guten Übergang !

**Comment by ChrisWUP** [comment]
Das liest hier keiner, richtig ?

**Reply by ollyfr** [reply]
eigentlich schon :) aber nicht jeder kann was ändern

**Reply by ChrisWUP** [reply]
Danke für die Rückmeldung :-)
Hast Du eine Möglichkeit gefunden die 6 Bereiche die in der Readme benannt werden zu konfigurieren oder eine Idee was ich eingeben muss damit sich beim Klick auf das Widget die App öffnet ?

**Reply by ollyfr** [reply]
nein ich bin selber nicht so bewandelt hier viel anzupassen, habe bei mir nur meine Jährlichen Leasing Km (Hochrechnung nach täglich gefahrenen Km auf ein Jahr) eingebaut.

**Reply by dschablowsky** [reply]
@ollyfr Benötigst du die Leasing Km noch für dein separates Widget oder reicht es dir in diesem?

**Reply by ollyfr** [reply]
Ich habe es für meine Belange angepasst 


![254429CC-0337-4E26-B8BF-0AC71D13E3F0](https://user-images.githubusercontent.com/61087107/103918975-c700a680-510f-11eb-92af-945d27f157d2.jpeg)

**Reply by dschablowsky** [reply]
Wie hast du denn die Hochrechnung berechnet? Nimmst du einfach die bisher gefahrenen Km und die dafür benötigten Tage und rechnest das auf die volle Leasingzeit hoch?

**Comment by dschablowsky** [comment]
Hi Chris, sorry für die späte Antwort. Ich habe die Diskussions-Funktion erst vor kurzem aktiviert und kannte diese Funktion noch nicht.

Jetzt zu deinen Anmerkungen:
1.) Sobald das Widget auf deinem Homescreen ist, musst du lange darauf drücken bis das Popup-Menü kommt, wo du "Widget bearbeiten" auswählen musst. Da wählst du bei "When Interacting" die "Open URL" Option aus und tippst bei URL ein "fordapp://". Damit wird die FordPass App geöffnet. 
Kleiner Hinweis: die normale und die Pro Version von FordPass haben die selbe URL. Bei mir wird die Pro Version geöffnet. Es sollte also am besten nur eine Version installiert sein. Ansonsten kann man nicht vorhersehen, welche App geöffnet wird.
2.) Die Konfiguration ist bisher noch nicht implementiert. Ich hatte Privat einfach viel zu viel zu tun und kam leider noch nicht dazu. Ich werde mich aber so bald wie möglich da dransetzen - wenn alles klappt vielleicht sogar noch diese Woche

**Reply by ChrisWUP** [reply]
Hi Damian,

hey, schön von Dir zu hören, klasse Widget, danke für Deine Antworten.
Privat geht vor, kein Stress ... 

Hast Du die SUPER Anmerkung von "ollyfr" gelesen ?
Eine Hochrechnung der Leasing Kilometer wäre echt eine coole Idee.

Dir ein frohes neues, DANKE für Deine Arbeit und Dein Feedback :-)

**Reply by ChrisWUP** [reply]
NACHTRAG: Wenn ich die URL: fordapp:// eingebe dann öffnet sich zwar (eine) App aber der Bildschirm bleibt komplett schwarz.
Ich habe die "normale" Ford Pass app drauf. Eine Idee dazu ?

**Reply by ChrisWUP** [reply]
2ter Nachtrag:
Habe mir jetzt die PRO App heruntergeladen und installiert, damit klappt es !
Normale App gelöscht :-)

**Reply by dschablowsky** [reply]
Ich benutze auch die Pro App. Die läuft schneller uns zieht mir nicht Akku nicht leer ;-)

**Comment by ollyfr** [comment]
Ja so in etwa

Ich gebe das Datum im Script ein an dem ich das Auto abhole

Die km aus der App
Geteilt durch die vergangen Tage mal 365

Habe es etwas kompliziert geschrieben aber funktioniert

Ich sende dir das Skript zu

Mit freundlichen Grüssen
Oliver Fischer
Am 7. Jan. 2021, 18:02 +0100 schrieb Damian Schablowsky <notifications@github.com>:

**Reply by ChrisWUP** [reply]
Das ist echt eine gute Idee gewesen, Oliver :-)

**Reply by ollyfr** [reply]
Ist es auch möglich den Reifendruck mit ins Bild zu bekommen? In meinem Script sind ja die 4 Fenster separat sichtbar

**Reply by ollyfr** [reply]
Wo kann ich die Liste sehen was alles vom Server heruntergeladen wird, wo wird diese abgelegt ?
viele Grüsse olly

**Reply by dschablowsky** [reply]
Es wäre super, wenn du mir das Skript zusenden kannst :-)

Das ist kein Problem. Die Daten kann man an folgender Stelle sehen:
Füge in der Funktion `async function fetchCarData()` nach der Zeile `let vehicleStatus = rawData.vehiclestatus` folgenden Code ein:
`console.log(vehicleStatus)`
Die Ausgabe ich leider als einzelner String. Du kannst die Ausgabe aber kopieren und auf der Seite https://www.freeformatter.com/json-formatter.html in ein leserliches Format bringen. Da dürfte auch unter anderem der Reifendruck stehen (nicht in Bar!)

**Reply by dschablowsky** [reply]
Ich habe die berechnete Distanz pro Jahr sowie den Luftdruck im neuen Update eingebaut ;-)

**Comment by ollyfr** [comment]
Super

Ich habe nur leider eine. Fehler

Mit freundlichen Grüssen
Oliver Fischer
Am 22. Feb. 2021, 12:30 +0100 schrieb Damian Schablowsky <notifications@github.com>:

**Reply by dschablowsky** [reply]
Was für einen Fehler hast du denn?

**Comment by ollyfr** [comment]
2021-02-22 13:13:05: FM: read local data
2021-02-22 13:13:05: FM: read local data
2021-02-22 13:13:05: 6875
2021-02-22 13:13:05: 126.59242462962963
2021-02-22 13:13:05: 126.59242462962963
2021-02-22 13:13:05: 19822
2021-02-22 13:13:05: Error on line 898:40: TypeError: undefined is not an object (evaluating 'carData.tirePressure['leftFront']')

Mit freundlichen Grüssen
Oliver Fischer
Am 22. Feb. 2021, 13:16 +0100 schrieb Damian Schablowsky <notifications@github.com>:

**Reply by dschablowsky** [reply]
Hast du etwas am Code geändert? Und hat dein Auto RDKS?

**Comment by ollyfr** [comment]
ich habe nur meine KOntaktdaten eingegeben und das Auto ist von 09/2020 also mit RDKs

 


 

vg Olly

 
 

Gesendet: Montag, 22. Februar 2021 um 13:19 Uhr
Von: "Damian Schablowsky" <notifications@github.com>
An: "dschablowsky/FordPassWidget" <FordPassWidget@noreply.github.com>
Cc: "ollyfr" <oliver@fischer-me.de>, "Mention" <mention@noreply.github.com>
Betreff: Re: [dschablowsky/FordPassWidget] FordPass App öffnen bei "Klick" ? (#4)


 

Hast du etwas am Code geändert? Und hat dein Auto RDKS?

**Reply by dschablowsky** [reply]
kannst du mir bitte deinen Quellcode hier hin OHNE Kontaktdaten hochladen?
https://pastebin.com/
Dann kann ich es mir nochmal genauer anschauen

**Reply by dschablowsky** [reply]
Und mir bitte den link schicken

**Reply by dschablowsky** [reply]
Zudem wäre es noch gut, wenn du mir die Daten schickst, die du vom Server bekommst.
Dafür bitte im Quellcode `debug = true` und `clearFileManagerOnNextRun = true` setzen und aus der Konsole den Teil ab `{"vehiclestatus":` , aber OHNE VIN schicken. Vielleicht werden die Daten von deinem Auto anders verarbeitet und abgespeichert

**Comment by ollyfr** [comment]
das Script ist anbei 

 

vom Handy komme ich nicht auf den Pfad
 

vg
Olly

 
 

Gesendet: Montag, 22. Februar 2021 um 13:24 Uhr
Von: "Damian Schablowsky" <notifications@github.com>
An: "dschablowsky/FordPassWidget" <FordPassWidget@noreply.github.com>
Cc: "ollyfr" <oliver@fischer-me.de>, "Mention" <mention@noreply.github.com>
Betreff: Re: [dschablowsky/FordPassWidget] FordPass App öffnen bei "Klick" ? (#4)


 

kannst du mir bitte deinen Quellcode hier hin OHNE Kontaktdaten hochladen?
https://pastebin.com/
Dann kann ich es mir nochmal genauer anschauen

**Comment by ollyfr** [comment]
komisch jetz geht es 

 

naja falls es nochmal spint melde ich mich

 

 
 

Gesendet: Montag, 22. Februar 2021 um 13:28 Uhr
Von: "Damian Schablowsky" <notifications@github.com>
An: "dschablowsky/FordPassWidget" <FordPassWidget@noreply.github.com>
Cc: "ollyfr" <oliver@fischer-me.de>, "Mention" <mention@noreply.github.com>
Betreff: Re: [dschablowsky/FordPassWidget] FordPass App öffnen bei "Klick" ? (#4)


 

Zudem wäre es noch gut, wenn du mir die Daten schickst, die du vom Server bekommst.
Dafür bitte im Quellcode debug = true und clearFileManagerOnNextRun = true setzen und aus der Konsole den Teil ab {"vehiclestatus": , aber OHNE VIN schicken. Vielleicht werden die Daten von deinem Auto anders verarbeitet und abgespeichert

**Reply by dschablowsky** [reply]
Ich glaube ich kenne den Fehler:
Das Skript aktualisiert frühestens nach 5 Minuten. Die Daten waren wahrscheinlich noch nicht aktualisiert und es wurde mit den lokalen Daten gearbeitet, die den Reifendruck noch nicht hatten. Nach den 5 Minuten wurden die neusten Daten gezogen und dann angezeigt

**Comment by ollyfr** [comment]
Ja schein so, habe das andere Script geschlossen
 

Mit freundlichen Grüssen
Oliver Fischer

liebe Grüsse
Olly

 
 

Gesendet: Montag, 22. Februar 2021 um 13:41 Uhr
Von: "Damian Schablowsky" <notifications@github.com>
An: "dschablowsky/FordPassWidget" <FordPassWidget@noreply.github.com>
Cc: "ollyfr" <oliver@fischer-me.de>, "Mention" <mention@noreply.github.com>
Betreff: Re: [dschablowsky/FordPassWidget] FordPass App öffnen bei "Klick" ? (#4)


 

Ich glaube ich kenne den Fehler:
Das Skript aktualisiert frühestens nach 5 Minuten. Die Daten waren wahrscheinlich noch nicht aktualisiert und es wurde mit den lokalen Daten gearbeitet, die den Reifendruck noch nicht hatten. Nach den 5 Minuten wurden die neusten Daten gezogen und dann angezeigt
